packages/sdk-metrics/src/ObservableResult.ts

```diff
@@ -41,6 +41,12 @@ export class ObservableResultImpl implements ObservableResult {
    * Observe a measurement of the value associated with the given attributes.
    */
   observe(value: number, attributes: MetricAttributes = {}): void {
+    if (typeof value !== 'number') {
```

Comment:
Can we try to cast to a number first before returning?

Reply:
Hmm, I would prefer just to accept actual numbers. :thinking: The metrics SDK is already very complex, and once we support this, we may only be able to walk this back by breaking users that rely on this behavior.

Reply:
IMO, implicit type coercion could be a source of bugs and the result can be surprising, e.g. 
- https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#non-number_delay_values_are_silently_coerced_into_numbers
- https://docs.google.com/presentation/d/1m5R5J98W6adegghgkAlbSuFgAYJDT52yyFVdAqLjm00/edit#slide=id.g13a69787e9f_0_0

Reply:
@JamieDanielson do you think your concern has been addressed yet?

Reply:
I'm convinced, thanks for the feedback!